easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -83,6 +95,10 @@ def configure_step(self, srcdir=None, builddir=None):
                 srcdir = default_srcdir
 
         options = ['-DCMAKE_INSTALL_PREFIX=%s' % self.installdir]
+
+        if '-DCMAKE_BUILD_TYPE' not in self.cfg['configopts']:
+            options.append("-DCMAKE_BUILD_TYPE=%s" % self.cfg['buildtype'])
```

Comment:
people may get confused where it comes from, and why the `build_type` they specify is totally ignored...

separate build dir or not is less of an issue, since it doesn't affect the end result (it'll either work, or it won't)

imagine an easyblock deriving from `CMakeMake` updating `configopts` to include `-DCMAKE_BUILD_TYPE`, for example; clear log messages will help in figuring that out